cmd/patch-release-notification/main.go

```diff
@@ -0,0 +1,318 @@
+		}, fmt.Errorf("parsing template: %w", err)
+	}
+
+	var tmplBytes bytes.Buffer
+	err = tmpl.Execute(&tmplBytes, output)
+	if err != nil {
+		return events.APIGatewayProxyResponse{
+			Body:       `{"status": "nok"}`,
+			StatusCode: http.StatusInternalServerError,
+		}, fmt.Errorf("parsing values to the template: %w", err)
+	}
+
+	if !shouldSendEmail {
+		logrus.Info("No email is needed to send")
+		return events.APIGatewayProxyResponse{
+			Body:       `{"status": "ok"}`,
+			StatusCode: http.StatusOK,
+		}, nil
+	}
+
+	logrus.Info("Sending mail")
+	subject := "[Please Read] Patch Releases cherry-pick deadline"
+	fromEmail := o.Config.FromEmail
+
+	recipient := Recipient{
+		toEmails: []string{o.Config.ToEmail},
+	}
+
+	if !o.Config.NoMock {
+		logrus.Info("This is a mock only, will print out the email before sending to a test mailing list")
+		fmt.Println(tmplBytes.String())
+		// if is a mock we send the email to ourselves to test
+		recipient.toEmails = []string{o.Config.FromEmail}
+	}
+
+	err = o.SendEmailRawSES(tmplBytes.String(), subject, fromEmail, recipient)
+	if err != nil {
+		return events.APIGatewayProxyResponse{
+			Body:       `{"status": "nok"}`,
+			StatusCode: http.StatusInternalServerError,
+		}, fmt.Errorf("parsing values to the template: %w", err)
+	}
+
+	return events.APIGatewayProxyResponse{
+		Body:       `{"status": "ok"}`,
+		StatusCode: 200,
+	}, nil
+}
+
+// Recipient struct to hold email IDs
+type Recipient struct {
+	toEmails  []string
+	ccEmails  []string
+	bccEmails []string
+}
+
+// SendEmailSES sends email to specified email IDs
+func (o *Options) SendEmailRawSES(messageBody, subject, fromEmail string, recipient Recipient) error {
+	// create raw message
+	msg := gomail.NewMessage()
+
+	// set to section
+	recipients := make([]*string, 0, len(recipient.toEmails))
+	for _, r := range recipient.toEmails {
+		recipient := r
+		recipients = append(recipients, &recipient)
+	}
+
+	// Set to emails
+	msg.SetHeader("To", recipient.toEmails...)
+
+	// cc mails mentioned
+	if len(recipient.ccEmails) != 0 {
+		// Need to add cc mail IDs also in recipient list
+		for _, r := range recipient.ccEmails {
+			recipient := r
+			recipients = append(recipients, &recipient)
+		}
+		msg.SetHeader("cc", recipient.ccEmails...)
+	}
+
+	// bcc mails mentioned
+	if len(recipient.bccEmails) != 0 {
+		// Need to add bcc mail IDs also in recipient list
+		for _, r := range recipient.bccEmails {
+			recipient := r
+			recipients = append(recipients, &recipient)
+		}
+		msg.SetHeader("bcc", recipient.bccEmails...)
+	}
+
+	// create an SES session.
+	svc := ses.New(o.AWSSess)
+
+	msg.SetAddressHeader("From", fromEmail, "Release Managers")
+	msg.SetHeader("To", recipient.toEmails...)
+	msg.SetHeader("Subject", subject)
+	msg.SetBody("text/html", messageBody)
+
+	// create a new buffer to add raw data
+	var emailRaw bytes.Buffer
+	_, err := msg.WriteTo(&emailRaw)
+	if err != nil {
+		log.Printf("failed to write mail: %v\n", err)
+		return err
```

Comment:
Similar as above, we should probably drop this `log` in favor of logrus or returning an error with that message.